R/write_timeseries_dsg.R

```diff
@@ -1,170 +1,182 @@
-#' @param add_to_existing \code{boolean} If TRUE and the file already exists, 
+#' @param add_to_existing \code{boolean} If TRUE and the file already exists,
 #' variables will be added to the existing file. See details for more.
 #' @param overwrite boolean error if file exists.
-#' 
+#'
 #' @description
 #' This function creates a timeseries discrete sampling geometry NetCDF file.
 #' It uses the orthogonal array encoding to write one \code{data.frame} per
 #' function call. This encoding is best suited to data with the same number of
 #' timesteps per instance (e.g. geometry or station).
-#' 
+#'
 #' @details
 #' Suggested Global Variables:
-#' c(title = "title", 
-#' abstract = "history", 
-#' provider site = "institution", 
-#' provider name ="source", 
+#' c(title = "title",
+#' abstract = "history",
+#' provider site = "institution",
+#' provider name ="source",
 #' description = "description")
-#' 
+#'
 #' Note regarding add_to_existing:
-#' add_to_existing = TRUE should only be used to add variables to an existing 
-#' NetCDF discrete sampling geometry file. All other inputs should be the 
-#' same as are already in the file. If the functions is called with 
-#' add_to_existing=FALSE (the default), it will overwrite an existing file 
-#' with the same name. The expected usage is to call this function repeatedly 
+#' add_to_existing = TRUE should only be used to add variables to an existing
+#' NetCDF discrete sampling geometry file. All other inputs should be the
+#' same as are already in the file. If the functions is called with
+#' add_to_existing=FALSE (the default), it will overwrite an existing file
+#' with the same name. The expected usage is to call this function repeatedly
 #' only changing the data, data_unit, data_prec and data_metadata inputs.
-#' 
+#'
 #' See the timeseries vignette for more information.
-#' 
+#'
 #' @references
 #' \enumerate{
 #'   \item \url{https://www.unidata.ucar.edu/software/netcdf-java/v4.6/reference/FeatureDatasets/CFpointImplement.html}
 #'   \item \url{http://cfconventions.org/cf-conventions/cf-conventions.html#_orthogonal_multidimensional_array_representation}
 #'   \item \url{http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/build/cf-conventions.html#time-series-data}
 #' }
-#' 
+#'
 #' @importFrom RNetCDF open.nc close.nc create.nc dim.def.nc var.def.nc var.put.nc att.put.nc
 #' @importFrom methods is
-#' 
+#'
 #' @export
-write_timeseries_dsg = function(nc_file, instance_names, lats, lons, times, 
+write_timeseries_dsg = function(nc_file, instance_names, lats, lons, times,
                                 data, alts=NA, data_unit='',
 																data_prec='double',
 																data_metadata=list(name='data',long_name='unnamed data'),
 																time_units = 'days since 1970-01-01 00:00:00',
+																instance_dim_name,
```

Comment:
While it is a little duplication, we should probably do `instance_dim_name = "instance"` and `dsg_timeseries_id = "instance_name"` rather than using the `missing` check below to stay consistent.